backend/src/schema/resolvers/posts.spec.js

```diff
@@ -682,58 +682,62 @@ describe('UpdatePost', () => {
       })
 
       describe('PostOrdering', () => {
-        let pinnedPost, admin
         beforeEach(async () => {
-          ;[pinnedPost] = await Promise.all([
-            neode.create('Post', {
-              id: 'im-a-pinned-post',
-              pinned: true,
-            }),
-            neode.create('Post', {
-              id: 'i-was-created-after-pinned-post',
-              createdAt: '2019-10-22T17:26:29.070Z', // this should always be 3rd
-            }),
-          ])
-          admin = await user.update({
-            role: 'admin',
-            name: 'Admin',
-            updatedAt: new Date().toISOString(),
+          await factory.create('Post', {
+            id: 'im-a-pinned-post',
+            createdAt: '2019-11-22T17:26:29.070Z',
+            pinned: true,
+          })
+          await factory.create('Post', {
+            id: 'i-was-created-before-pinned-post',
+            // fairly old, so this should be 3rd
+            createdAt: '2019-10-22T17:26:29.070Z',
           })
-          await admin.relateTo(pinnedPost, 'pinned')
         })
 
-        it('pinned post appear first even when created before other posts', async () => {
-          const postOrderingQuery = gql`
-            query($orderBy: [_PostOrdering]) {
-              Post(orderBy: $orderBy) {
-                id
-                pinnedAt
+        describe('order by `pinned_asc` and `createdAt_desc`', () => {
+          beforeEach(() => {
+            // this is the ordering in the frontend
+            variables = { orderBy: ['pinned_asc', 'createdAt_desc'] }
+          })
+
+          it('pinned post appear first even when created before other posts', async () => {
+            const postOrderingQuery = gql`
+              query($orderBy: [_PostOrdering]) {
+                Post(orderBy: $orderBy) {
+                  id
+                  pinned
+                  createdAt
+                  pinnedAt
+                }
               }
-            }
-          `
-          const expected = {
-            data: {
-              Post: [
-                {
-                  id: 'im-a-pinned-post',
-                  pinnedAt: expect.any(String),
-                },
-                {
-                  id: 'p9876',
-                  pinnedAt: null,
-                },
-                {
-                  id: 'i-was-created-after-pinned-post',
-                  pinnedAt: null,
-                },
-              ],
-            },
-            errors: undefined,
-          }
-          variables = { orderBy: ['pinned_desc', 'createdAt_desc'] }
```

Comment:
> [<img alt="roschaefer" height="40" width="40" align="left" src="https://avatars1.githubusercontent.com/u/38261864?s=88&v=4">](https://github.com/roschaefer) **Authored by [roschaefer](https://github.com/roschaefer)**
_<time datetime="2020-01-23T18:39:55Z" title="Thursday, January 23rd 2020, 7:39:55 pm +01:00">Jan 23, 2020</time>_

---
> **Outdated (history rewrite)** - original diff
---
```diff
@@ -682,58 +682,62 @@ describe('UpdatePost', () => {
       })
 
       describe('PostOrdering', () => {
-        let pinnedPost, admin
         beforeEach(async () => {
-          ;[pinnedPost] = await Promise.all([
-            neode.create('Post', {
-              id: 'im-a-pinned-post',
-              pinned: true,
-            }),
-            neode.create('Post', {
-              id: 'i-was-created-after-pinned-post',
-              createdAt: '2019-10-22T17:26:29.070Z', // this should always be 3rd
-            }),
-          ])
-          admin = await user.update({
-            role: 'admin',
-            name: 'Admin',
-            updatedAt: new Date().toISOString(),
+          await factory.create('Post', {
+            id: 'im-a-pinned-post',
+            createdAt: '2019-11-22T17:26:29.070Z',
+            pinned: true,
+          })
+          await factory.create('Post', {
+            id: 'i-was-created-before-pinned-post',
+            // fairly old, so this should be 3rd
+            createdAt: '2019-10-22T17:26:29.070Z',
           })
-          await admin.relateTo(pinnedPost, 'pinned')
         })
 
-        it('pinned post appear first even when created before other posts', async () => {
-          const postOrderingQuery = gql`
-            query($orderBy: [_PostOrdering]) {
-              Post(orderBy: $orderBy) {
-                id
-                pinnedAt
+        describe('order by `pinned_asc` and `createdAt_desc`', () => {
+          beforeEach(() => {
+            // this is the ordering in the frontend
+            variables = { orderBy: ['pinned_asc', 'createdAt_desc'] }
+          })
+
+          it('pinned post appear first even when created before other posts', async () => {
+            const postOrderingQuery = gql`
+              query($orderBy: [_PostOrdering]) {
+                Post(orderBy: $orderBy) {
+                  id
+                  pinned
+                  createdAt
+                  pinnedAt
+                }
               }
-            }
-          `
-          const expected = {
-            data: {
-              Post: [
-                {
-                  id: 'im-a-pinned-post',
-                  pinnedAt: expect.any(String),
-                },
-                {
-                  id: 'p9876',
-                  pinnedAt: null,
-                },
-                {
-                  id: 'i-was-created-after-pinned-post',
-                  pinnedAt: null,
-                },
-              ],
-            },
-            errors: undefined,
-          }
-          variables = { orderBy: ['pinned_desc', 'createdAt_desc'] }
```


This is different from what our frontend does

Reply:
> [<img alt="mattwr18" height="40" width="40" align="left" src="https://avatars1.githubusercontent.com/u/38261864?s=88&v=4">](https://github.com/mattwr18) **Authored by [mattwr18](https://github.com/mattwr18)**
_<time datetime="2020-01-23T18:40:46Z" title="Thursday, January 23rd 2020, 7:40:46 pm +01:00">Jan 23, 2020</time>_

---


off the top of my head, it seems to be exactly what it does, oder?

Reply:
> [<img alt="roschaefer" height="40" width="40" align="left" src="https://avatars1.githubusercontent.com/u/38261864?s=88&v=4">](https://github.com/roschaefer) **Authored by [roschaefer](https://github.com/roschaefer)**
_<time datetime="2020-01-23T18:50:57Z" title="Thursday, January 23rd 2020, 7:50:57 pm +01:00">Jan 23, 2020</time>_

---


https://github.com/Human-Connection/Human-Connection/blob/master/webapp/pages/index.vue#L157
![nope](https://media.giphy.com/media/fsPcMdeXPxSP6zKxCA/giphy.gif)